### PR TITLE
Add CAPQL algorithm

### PR DESCRIPTION
**Pull request by LucasAlegre**
Issue #42

### CODE REVIEW

File: morl_baselines/multi_policy/capql/capql.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+)
+
+
+LOG_SIG_MAX = 2
+LOG_SIG_MIN = -20
+EPSILON = 1e-6
+
+
+class ReplayMemory:
+    """Replay memory."""
```
Comment (ffelten) [inline morl_baselines/multi_policy/capql/capql.py:34]: Can't we use the implementation of replay buffer we have in the repo?

File: morl_baselines/multi_policy/capql/capql.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+        """Train the agent.
+
+        Args:
+            total_timesteps (int): Total number of timesteps to train the agent for.
+            eval_env (gym.Env): Environment to use for evaluation.
+            ref_point (np.ndarray): Reference point for hypervolume calculation.
+            known_pareto_front (Optional[List[np.ndarray]]): Optimal Pareto front, if known.
+            num_eval_weights_for_front (int): Number of weights to evaluate for the Pareto front.
+            num_eval_episodes_for_front: number of episodes to run when evaluating the policy.
+            weight_selection_algo (str): Weight selection algorithm to use.
```
Comment (ffelten) [inline morl_baselines/multi_policy/capql/capql.py:401]: these comments seem off

**Review by ffelten — CHANGES_REQUESTED** [review]
looks good except the comment

File: morl_baselines/multi_policy/capql/capql.py (hunk 1 above)
Comment (LucasAlegre) [reply morl_baselines/multi_policy/capql/capql.py:34]: They also push reward weights to the buffer. We could adapt our replay buffer in the future, but for now I think it is easier to just use the buffer in their original code.

File: morl_baselines/multi_policy/capql/capql.py (hunk 2 above)
Comment (LucasAlegre) [reply morl_baselines/multi_policy/capql/capql.py:401]: Good catch!

**Pull request merged**